sklearn/linear_model/sag_fast.pyx

```diff
@@ -191,87 +440,74 @@ def sag(SequentialDataset dataset,
                         with gil:
                             print("rescaling...")
                     scale_weights(weights, wscale, n_features, n_samples,
-                                  itr, cumulative_sums, feature_hist,
-                                  sum_gradient)
+                                  n_classes, sample_itr, cumulative_sums,
+                                  feature_hist, sum_gradient)
                     wscale = 1.0
 
-            if infinity:
-                break
-
             # check if the stopping criteria is reached
-            scale_weights(weights, wscale, n_features, n_samples, n_samples - 1,
-                          cumulative_sums, feature_hist, sum_gradient)
+            scale_weights(weights, wscale, n_features, n_samples, n_classes,
+                          n_samples - 1, cumulative_sums, feature_hist,
+                          sum_gradient)
             wscale = 1.0
-            
+
             max_change = 0.0
             max_weight = 0.0
-            for j in range(n_features):
-                max_weight = fmax(max_weight, fabs(weights[j]))
-            for j in range(n_features):
+            for feature_ind in range(n_features):
+                max_weight = fmax(max_weight, fabs(weights[feature_ind]))
+            for feature_ind in range(n_features):
```

Comment:
You could remove this line right to loop only once and get `max_weight` and `max_change` simultaneously.